packages/esm-appointments-app/src/form/appointments-form.component.tsx

```diff
@@ -953,10 +953,13 @@ function TimeAndDuration({ t, watch, control, services, errors }) {
               max={1440}
               min={0}
               onBlur={onBlur}
-              onChange={(event) => onChange(Number(event.target.value))}
+              onChange={(event) => {
+                const formattedValue = formatNumber(event.target.value);
+                onChange(Number(formattedValue));
```

Comment:
@Muppasanipraneeth In scenarios where the user attempts to clear the field, `event.target.value` is an empty string `""`; you probably want to nullify the empty string value:

```
onChange(event.target.value === '' ? null : Number(event.target.value));
```